src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportResourceLoader.cs

```diff
@@ -151,6 +151,11 @@ private ImportResource ParseImportRawContent(string resourceType, (string Conten
                 {
                     throw new FormatException("Resource type not match.");
                 }
+
+                if (!string.IsNullOrEmpty(resourceType) && resourceType.Equals("SearchParameter", StringComparison.Ordinal))
```

Comment:
We need to be checking `importResource.ResourceWrapper?.ResourceTypeName` here to see if the imported resource is a SearchParameter. The logic in the controller should handle this case.